### DISCUSSION TITLE
Vcc in a rule to send to Domoticz

### DISCUSSION THREAD

**Discussion by kk02067** [arendst/Tasmota#20662] [Support request] [answered]
I compile tasmota with,

#define USE_ADC_VCC

and I can see the voltage of the esp in the 8266 STATE topic.

But I dont know how I can use that value in RULES to publish to Domoticz and use there. 
Any tips please?

**Comment by barbudor** [comment]
May be you can get inspiration from the existing examples in the doc ? 
https://tasmota.github.io/docs/Rules/#use-of-variables-and-tele-in-domoticz

**Comment by kk02067** [comment]
I have looked at that but I dont know how that helps. The Vcc value appears not as a sensor but as a value in the state of the esp as  below.

19:52:08.423 MQT: tele/tasmota_5D8F86/STATE = {"Time":"2024-02-04T19:52:08","Uptime":"0T00:00:08","UptimeSec":8,"Vcc":3.078,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"Subnet AC","BSSId":"B0:4E:26:2E:35:74","Channel":11,"Mode":"11n","RSSI":60,"Signal":-70,"LinkCount":1,"Downtime":"0T00:00:03"}}

What I would like to do is someting like this

ON tele#vcc DO publish domoticz/in {"idx":513,"nvalue":0,"svalue":%Vcc%} ENDON"}

But that wont work. Maybe i am just understanding it wrong

Any more tips?

**Reply by barbudor** [reply]
2 errors in your attemp
- VCC should be on the same side as `tele`, and for those triggers that are at the root of the JSON, you need `#state` (not because it is in the STATE message but because it is at root of the JSON)
- The value is in `%value%`, not `%vcc%`
```
ON tele-VCC#state DO publish domoticz/in {"idx":513,"nvalue":0,"svalue":%value%} ENDON
```

**Reply by sfromis** [reply, marked as answer]
The syntax for JSON-based TelePeriod triggers should be `tele-Vcc`, without `#state`
(I tested this with `tele-LoadAvg` on the same level in the STATE JSON)

**Reply by kk02067** [reply]
Thanks a lot guys. That last one worked like I wanted.
